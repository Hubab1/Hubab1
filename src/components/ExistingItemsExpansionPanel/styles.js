import { css } from 'emotion';

export const existingItemRow = css`
    border-top: 1px solid #EEEEEE;
    padding: 10px 0;
    display: flex;
    text-align: left;
    justify-content: space-between;
`;

export const existingItemsContainer = css`
    margin-bottom: 15px;
    border-top: 1px solid #EEEEEE;
    border-bottom: 1px solid #EEEEEE;
    .MuiExpansionPanelSummary-root {
        padding: 0;
    }
`;
