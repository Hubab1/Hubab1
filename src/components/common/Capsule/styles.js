import {
    css
} from 'emotion';

export const container = css`
    margin-bottom: 10px;
    a {
        text-decoration: none;
    }
`

export const paperRoot = css`
    box-shadow: 0px 2px 4px 2px rgba(0,0,0,.1) !important;
    border-width: 0px !important;
    padding: 15px !important;
    font-size: 16px !important;
    border-radius: 5px;
    .MuiButton-root {
        font-weight: 600;
    }
`

export const prefix = css`
    font-size: 55px;
    height: 55px;
    line-height: 67px;
    width: 100px;
    text-align: center;
`

export const label = css`
    flex: 1;
    padding-left: 0;
    color: #454B57;
    font-weight: 400;
    display: flex;
    font-size: 16px;
`

export const contentContainer = css`
    display: flex;
    flex: 1;
    align-items: center;
    text-transform: none;
    padding-bottom: 16px;
`

export const buttonRoot = css`
    border-radius: 30px !important;
    text-transform: none !important;
    font-size: 16px !important;
`

export const anchor = css`
    position: relative;
    top: -90px;
`
