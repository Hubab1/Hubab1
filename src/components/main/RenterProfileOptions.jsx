import React from 'react';
import Button from '@material-ui/core/Button';
import Page from './Page';
import MultiSelectChoice from './MultiSelectChoice';

export default class RentalProfileOptions extends React.Component {
    render () {
        return (
            <Page>
                <div>
                    <MultiSelectChoice 
                        prefix="👪"
                        label="Other adults will live here"
                    />
                    <div style={{height: 8}}></div>
                    <MultiSelectChoice
                        prefix="🐶"
                        label="Pets will live here"
                    />
                    <div style={{height: 8}}></div>
                    <MultiSelectChoice
                        prefix="💰"
                        label="I'll need a guarantor"
                    />
                    <div style={{height: 8}}></div>
                    <MultiSelectChoice
                        prefix="🚗"
                        label="I'd like a parking space"
                    />
                    <div style={{height: 8}}></div>
                    <MultiSelectChoice
                        prefix="🛍️"
                        label="I'll need extra storage"
                    />
                    <div style={{height: 8}}></div>
                </div>
            </Page>
        );
    }
}