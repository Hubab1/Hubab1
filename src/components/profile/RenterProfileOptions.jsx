import React from 'react';
import Button from '@material-ui/core/Button';

import { Link } from 'assets/index';


export default class RentalProfileOptions extends React.Component {
    render () {
        return (
            <div>
                <div>
                    {['option1', 'option2', 'option3'].map((el, i) => (
                        <li key={i}>{el}</li>
                    ))}
                </div>
                <Button onClick={this.changeScreen} variant="contained" color="primary" type="submit" fullWidth>
                    {this.props.buttonText}
                </Button>
                <br/>
                <br/>
                <Link>Go Back</Link>
            </div>
        );
    }
}
