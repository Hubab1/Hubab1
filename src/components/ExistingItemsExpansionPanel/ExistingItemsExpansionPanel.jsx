import React from 'react';
import PropTypes from 'prop-types';

import pluralize from 'pluralize';
import {
    makeStyles,
    ExpansionPanel,
    ExpansionPanelSummary,
    ExpansionPanelDetails,
} from '@material-ui/core';
import { ExpandMore as ExpandMoreIcon } from '@material-ui/icons';

import { existingItemsContainer, existingItemRow } from './styles';

const useStyles = makeStyles(() => ({
    root: {
        display: 'block',
        padding: 0
    },
}));

function ExistingItemsExpansionPanel({children, defaultExpanded, label, labelQuantity}) {
    const classes = useStyles();
    labelQuantity = labelQuantity ?? 0;
    if (!labelQuantity) return null;
    const toggleLabel = `${labelQuantity} ${pluralize(label, labelQuantity)}`;
    return (
        <div className={existingItemsContainer}>
            <ExpansionPanel elevation={0} defaultExpanded={defaultExpanded}>
                <ExpansionPanelSummary
                    expandIcon={<ExpandMoreIcon />}
                >
                    {toggleLabel}&nbsp;Added
                </ExpansionPanelSummary>
                <ExpansionPanelDetails classes={{root:classes.root}}>
                    {React.Children.map(children, child => (
                        <div className={existingItemRow}>{child}</div>
                    ))}
                </ExpansionPanelDetails>
            </ExpansionPanel>
        </div>
    );
}

ExistingItemsExpansionPanel.propTypes = {
    children: PropTypes.array,
    label: PropTypes.string,
    labelQuantity: PropTypes.number,
    defaultExpanded: PropTypes.bool,
};

export default ExistingItemsExpansionPanel;
