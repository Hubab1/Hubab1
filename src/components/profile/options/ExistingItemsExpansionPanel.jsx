import React from 'react';
import PropTypes from 'prop-types';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import { makeStyles } from '@material-ui/core/styles';
import pluralize from 'pluralize';

import { existingItemsContainer, existingItemRow } from './styles';


const useStyles = makeStyles(theme => ({
    root: {
        display: 'block',
        padding: 0
    },
}));

function ExistingItemsExpansionPanel({children, label, labelQuantity}) {
    const classes = useStyles();
    const toggleLabel = `${labelQuantity} ${pluralize(label, labelQuantity)}`;
    return (
        <div className={existingItemsContainer}>
            <ExpansionPanel elevation={0}>
                <ExpansionPanelSummary
                    expandIcon={<ExpandMoreIcon />}
                >
                    {toggleLabel}&nbsp;Added
                </ExpansionPanelSummary>
                <ExpansionPanelDetails classes={{root:classes.root}}>
                    {React.Children.map(children, child => (
                        <div className={existingItemRow}>{child}</div>
                    ))}
                </ExpansionPanelDetails>
            </ExpansionPanel>
        </div>
    )
}

ExistingItemsExpansionPanel.propTypes = {
    children: PropTypes.array,
    label: PropTypes.string,
    labelQuantity: PropTypes.number,
};

export default ExistingItemsExpansionPanel;