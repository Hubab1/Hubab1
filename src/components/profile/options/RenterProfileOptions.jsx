import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { connect } from 'react-redux';

import ActionButton from 'components/common/ActionButton/ActionButton';
import { ROUTES, SORTED_CONFIG_OPTIONS } from 'app/constants';
import { updateRenterProfile } from 'reducers/renter-profile';
import RenterProfileListItem from './RenterProfileListItem';
import { H1, H3 } from 'assets/styles';
import withRelativeRoutes from 'app/withRelativeRoutes';
import guarantor from 'assets/images/guarantor.png';
import coapplicants from 'assets/images/coapplicants.png';
import doggie from 'assets/images/doggie.png';
import { BackLink } from 'components/common/BackLink';


const SkinnyH1 = styled(H1)`
    width: 70%;
`

const SpacedH3 = styled(H3)`
    margin-top: 15px;
    margin-bottom: 30px;
`

const optionConfig = {
    co_applicants: {
        prefix: <img alt="coapplicants" src={coapplicants}></img>,
        name: 'co_applicants',
        label: 'Other adults will live here',
        buttonLabel: 'Invite a roommate',
        route: ROUTES.CO_APPLICANTS
    },
    pets: {
        prefix: <img alt="dog" src={doggie}></img>,
        name: 'pets',
        label: 'Pets will live here',
        buttonLabel: 'Add a pet',
        route: ROUTES.PETS
    },
    guarantor: {
        prefix: <img alt="coins" src={guarantor}></img>,
        name: 'guarantor',
        label: 'I\'ll need a guarantor',
        buttonLabel: 'Invite a guarantor',
        route: ROUTES.GUARANTOR
    },
    parking: {
        prefix: '🚗',
        name: 'parking',
        label: 'I\'d like a parking space',
        buttonLabel: 'Add parking',
        route: ROUTES.PARKING
    },
    storage: {
        prefix: '🛍️',
        name: 'storage',
        label: 'I\'ll need extra storage',
        buttonLabel: 'Add storage',
        route: ROUTES.STORAGE
    }
}

export class RentalProfileOptions extends React.Component {
    onContinue = () => {
        // TODO: API call that makes some milestone?
        this.props._nextRoute();
    }



    render () {
        if (this.props.config == null || this.props.profile == null) return null;
        const unsortedOptionsSet = new Set(Object.keys(this.props.config.options).map(key => key.toLowerCase()));
        // for now we assume all require co_appplicants and guarantor... might want to do something smarter with RentalOptions in the future, but okie for now
        const sortedRentalOptions = ['co_applicants', 'guarantor'].concat(SORTED_CONFIG_OPTIONS.filter(option => unsortedOptionsSet.has(option))
        );

        return (
            <Fragment>
                <SkinnyH1>Let's Set Up Your Rental Profile</SkinnyH1>
                <SpacedH3>Complete the sections that apply to you and skip the ones that don't.</SpacedH3>
                <div>
                    {sortedRentalOptions.map(option => (
                        <RenterProfileListItem
                            key={option}
                            {...optionConfig[option]}
                        />
                    ))}
                </div>
                <ActionButton onClick={this.onContinue} marginTop={60} marginBottom={27}>Continue</ActionButton>
                <BackLink to={this.props._prev}/>
            </Fragment>
        );
    }
}

RentalProfileOptions.propTypes = {
    updateRenterProfile: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
    config: state.configuration,
    profile: state.renterProfile,
})

export default connect(mapStateToProps, {updateRenterProfile})(withRelativeRoutes(RentalProfileOptions, ROUTES.PROFILE_OPTIONS));
