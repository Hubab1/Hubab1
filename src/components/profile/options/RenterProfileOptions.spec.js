import React from 'react';
import { shallow } from 'enzyme';

import ExistingItemsExpansionPanel from 'components/profile/options/ExistingItemsExpansionPanel';
import Capsule from 'components/common/Capsule/Capsule';
import { RentalProfileOptions } from './RenterProfileOptions';
import mockConfig from 'reducers/mock-config.json';
import mockProfile from 'reducers/mock-profile.json';
import { RENTER_PROFILE_TYPE_PARKING } from 'app/constants';
import GenericFormDetail from 'components/common/GenericFormMessage';

let defaultProps;

beforeEach(() => {
    defaultProps = {
        updateRenterProfile: jest.fn(),
        profile: mockProfile,
        config: mockConfig,
        _nextRoute: jest.fn(),
        pageComplete: jest.fn().mockResolvedValue({}),
        location: { hash: '' },
    };
});

it('renders a Capsule component for each option in config.rental_options', function () {
    const wrapper = shallow(<RentalProfileOptions {...defaultProps} />);
    expect(wrapper.find(Capsule).length).toEqual(6);
});
it('renders GenericFormDetail if hasError=true', function () {
    const wrapper = shallow(<RentalProfileOptions {...defaultProps} />);
    wrapper.setState({ hasError: true });
    expect(wrapper.find(GenericFormDetail).length).toEqual(1);
});

it('matches snapshot', () => {
    const wrapper = shallow(<RentalProfileOptions {...defaultProps} />);
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('renders rental_option_config choices in correct order', () => {
    const wrapper = shallow(<RentalProfileOptions {...defaultProps} config={mockConfig} />);

    expect(wrapper.find(Capsule).first().props()['name']).toEqual('co_applicants');
    expect(wrapper.find(Capsule).at(1).props()['name']).toEqual('guarantor');
    expect(wrapper.find(Capsule).at(2).props()['name']).toEqual('pets');
    expect(wrapper.find(Capsule).at(3).props()['name']).toEqual('parking');
    expect(wrapper.find(Capsule).at(4).props()['name']).toEqual('storage');
    expect(wrapper.find(Capsule).at(5).props()['name']).toEqual('wine-cooler');
});

it('renders ExistingItemsExpansionPanel with coapplicants as existing items', () => {
    const wrapper = shallow(<RentalProfileOptions {...defaultProps} />);
    const coApplicantsProps = wrapper.find(Capsule).first().props();
    const coApplicantsWrapper = shallow(<Capsule {...coApplicantsProps} />);
    const coApplicantsPanel = coApplicantsWrapper.find(ExistingItemsExpansionPanel);
    expect(coApplicantsPanel.props()['children'].length).toEqual(3);
    expect(coApplicantsPanel.props()['children']).toMatchSnapshot();

    const parkingProps = wrapper.find(Capsule).at(3).props();
    const parkingWrapper = shallow(<Capsule {...parkingProps} />);
    expect(parkingWrapper.find(ExistingItemsExpansionPanel).length).toEqual(0);
});

it('renders ExistingItemsExpansionPanel for parking when there are selected_rental_options with quantity greater than 0', () => {
    const selectedParking = {
        parking: [
            {
                leasing_context: {},
                rental_option: { id: 102 },
                quoted_fee_amount: null,
                quoted_monthly_amount: '55.00',
                quoted_deposit_amount: null,
                id: 19002,
                quantity: 2,
            },
            {
                leasing_context: {},
                rental_option: { id: 63 },
                quoted_fee_amount: null,
                quoted_monthly_amount: '75.00',
                quoted_deposit_amount: null,
                id: 19002,
                quantity: 0,
            },
        ],
    };
    const selectedParkingProfile = Object.assign({}, mockProfile, { selected_rental_options: selectedParking });
    const wrapper = shallow(<RentalProfileOptions {...defaultProps} profile={selectedParkingProfile} />);
    const parkingProps = wrapper.find(Capsule).at(3).props();
    const parkingWrapper = shallow(<Capsule {...parkingProps} />);

    expect(parkingWrapper.find(ExistingItemsExpansionPanel).length).toEqual(1);
    expect(wrapper.getElement()).toMatchSnapshot();

    const parkingPanel = parkingWrapper.find(ExistingItemsExpansionPanel);
    expect(parkingPanel.props()['children'].length).toEqual(1);
    expect(parkingPanel.props()['children'][0]).toMatchSnapshot();
});

it('passes expansion panel defaultExpanded=True if anchor hash for rental option matches', () => {
    const selectedParking = {
        parking: [
            {
                leasing_context: {},
                rental_option: { id: 102 },
                quoted_fee_amount: null,
                quoted_monthly_amount: '55.00',
                quoted_deposit_amount: null,
                id: 19002,
                quantity: 2,
            },
        ],
    };
    const selectedParkingProfile = Object.assign({}, mockProfile, { selected_rental_options: selectedParking });

    const wrapper = shallow(
        <RentalProfileOptions
            {...defaultProps}
            profile={selectedParkingProfile}
            location={{ hash: `#${RENTER_PROFILE_TYPE_PARKING}` }}
        />
    );

    const coApplicantsProps = wrapper.find(Capsule).first().props();
    const coApplicantsWrapper = shallow(<Capsule {...coApplicantsProps} />);
    const coApplicantsPanel = coApplicantsWrapper.find(ExistingItemsExpansionPanel);
    expect(coApplicantsPanel.props()['defaultExpanded']).toEqual(false);

    const parkingProps = wrapper.find(Capsule).at(3).props();
    const parkingWrapper = shallow(<Capsule {...parkingProps} />);
    const parkingPanel = parkingWrapper.find(ExistingItemsExpansionPanel);
    expect(parkingPanel.props()['defaultExpanded']).toEqual(true);
});

describe('onContinue', function () {
    it('marks page complete on continue and calls next route', function () {
        const wrapper = shallow(<RentalProfileOptions {...defaultProps} />);
        return wrapper
            .instance()
            .onContinue()
            .then(() => {
                expect(defaultProps.pageComplete).toHaveBeenCalledWith('renter_profile');
                expect(defaultProps._nextRoute).toHaveBeenCalled();
            });
    });
});
