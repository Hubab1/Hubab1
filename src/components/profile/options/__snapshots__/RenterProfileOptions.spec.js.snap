// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Let's Set Up Your Rental Profile
  </ForwardRef(render)>
  <ForwardRef(render)>
    Complete the sections that apply to you and skip the ones that don't.
  </ForwardRef(render)>
  <div>
    <Capsule
      buttonLabel="Invite another roommate"
      expansionPanel={
        <ExistingItemsExpansionPanel
          defaultExpanded={false}
          label="Roommates"
          labelQuantity={2}
        >
          <ExistingRoommate
            item={
              Object {
                "first_name": "kreebs",
                "id": 71,
                "is_registered": false,
                "last_milestone": null,
                "last_name": "mcgreebs",
                "phone_number": "(383) 838-4849",
                "status": "Not Started",
              }
            }
            type="co_applicants"
          />
          <ExistingRoommate
            item={
              Object {
                "first_name": "jerry",
                "id": 73,
                "is_registered": true,
                "last_milestone": Object {
                  "created_at": "2020-07-23 15:22:49",
                  "event": 210,
                  "id": 10557,
                },
                "last_name": "maguire",
                "phone_number": "(333) 888-4449",
                "status": "Completed",
              }
            }
            type="co_applicants"
          />
        </ExistingItemsExpansionPanel>
      }
      label="I'll be living with roommates"
      limitReached={false}
      name="co_applicants"
      prefix={
        <img
          alt="coapplicants"
          src="coapplicants.png"
        />
      }
      route="/rental-profile/invite-roommates"
    />
    <Capsule
      buttonLabel="Invite a guarantor"
      expansionPanel={
        <ExistingItemsExpansionPanel
          defaultExpanded={false}
          label="Guarantor"
          labelQuantity={1}
        >
          <ExistingRoommate
            item={
              Object {
                "first_name": "elvish",
                "id": 74,
                "is_registered": false,
                "last_milestone": Object {
                  "created_at": "2020-07-23 15:22:49",
                  "event": 220,
                  "id": 10554,
                },
                "last_name": "parsley",
                "phone_number": "(222) 111-0000",
              }
            }
            type="guarantor"
          />
        </ExistingItemsExpansionPanel>
      }
      label="I'll need a guarantor"
      limitReached={true}
      name="guarantor"
      prefix={
        <img
          alt="coins"
          src="guarantor.png"
        />
      }
      route="/rental-profile/invite-guarantor"
      tip="This is a person that agrees to be legally responsible for the apartment, its condition, and the money owed for rent if you are unable to pay."
    />
    <Capsule
      buttonLabel="Add a pet"
      expansionPanel={false}
      label="I'll be living with pets"
      limitReached={false}
      name="pets"
      prefix={
        <img
          alt="dog"
          src="cat.svg"
        />
      }
      route="/rental-profile/pets"
    />
    <Capsule
      buttonLabel="Add Parking"
      expansionPanel={false}
      label="I'll need parking"
      limitReached={false}
      name="parking"
      prefix={
        <img
          alt="car parking"
          src="addparking.svg"
        />
      }
      route="/rental-profile/parking"
    />
    <Capsule
      buttonLabel="Add Storage"
      expansionPanel={false}
      label="I'll need storage"
      limitReached={false}
      name="storage"
      prefix={
        <img
          alt="storage"
          src="addstorage.svg"
        />
      }
      route="/rental-profile/storage"
    />
  </div>
  <ActionButton
    color="primary"
    disabled={false}
    marginBottom={27}
    marginTop={60}
    onClick={[Function]}
    successGreen={false}
    variant="contained"
  >
    Continue
  </ActionButton>
  <BackLink
    to="/"
  />
</React.Fragment>
`;

exports[`renders ExistingItemsExpansionPanel for parking when there are selected_rental_options with quantity greater than 0 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Let's Set Up Your Rental Profile
  </ForwardRef(render)>
  <ForwardRef(render)>
    Complete the sections that apply to you and skip the ones that don't.
  </ForwardRef(render)>
  <div>
    <Capsule
      buttonLabel="Invite another roommate"
      expansionPanel={
        <ExistingItemsExpansionPanel
          defaultExpanded={false}
          label="Roommates"
          labelQuantity={2}
        >
          <ExistingRoommate
            item={
              Object {
                "first_name": "kreebs",
                "id": 71,
                "is_registered": false,
                "last_milestone": null,
                "last_name": "mcgreebs",
                "phone_number": "(383) 838-4849",
                "status": "Not Started",
              }
            }
            type="co_applicants"
          />
          <ExistingRoommate
            item={
              Object {
                "first_name": "jerry",
                "id": 73,
                "is_registered": true,
                "last_milestone": Object {
                  "created_at": "2020-07-23 15:22:49",
                  "event": 210,
                  "id": 10557,
                },
                "last_name": "maguire",
                "phone_number": "(333) 888-4449",
                "status": "Completed",
              }
            }
            type="co_applicants"
          />
        </ExistingItemsExpansionPanel>
      }
      label="I'll be living with roommates"
      limitReached={false}
      name="co_applicants"
      prefix={
        <img
          alt="coapplicants"
          src="coapplicants.png"
        />
      }
      route="/rental-profile/invite-roommates"
    />
    <Capsule
      buttonLabel="Invite a guarantor"
      expansionPanel={
        <ExistingItemsExpansionPanel
          defaultExpanded={false}
          label="Guarantor"
          labelQuantity={1}
        >
          <ExistingRoommate
            item={
              Object {
                "first_name": "elvish",
                "id": 74,
                "is_registered": false,
                "last_milestone": Object {
                  "created_at": "2020-07-23 15:22:49",
                  "event": 220,
                  "id": 10554,
                },
                "last_name": "parsley",
                "phone_number": "(222) 111-0000",
              }
            }
            type="guarantor"
          />
        </ExistingItemsExpansionPanel>
      }
      label="I'll need a guarantor"
      limitReached={true}
      name="guarantor"
      prefix={
        <img
          alt="coins"
          src="guarantor.png"
        />
      }
      route="/rental-profile/invite-guarantor"
      tip="This is a person that agrees to be legally responsible for the apartment, its condition, and the money owed for rent if you are unable to pay."
    />
    <Capsule
      buttonLabel="Add a pet"
      expansionPanel={false}
      label="I'll be living with pets"
      limitReached={false}
      name="pets"
      prefix={
        <img
          alt="dog"
          src="cat.svg"
        />
      }
      route="/rental-profile/pets"
    />
    <Capsule
      buttonLabel="Manage Parking"
      expansionPanel={
        <ExistingItemsExpansionPanel
          defaultExpanded={false}
          label="Parking Space"
          labelQuantity={2}
        >
          <ExistingParkingOrStorage
            quantity={2}
            rentalOption={
              Object {
                "deposit_amount": null,
                "fee_amount": null,
                "id": 102,
                "included": 1,
                "limit": 6,
                "mandatory": true,
                "monthly_amount": "75.00",
                "name": "Garage (12x9)",
                "payment_time": 10,
                "type": 10,
              }
            }
          />
        </ExistingItemsExpansionPanel>
      }
      label="I'll need parking"
      limitReached={false}
      name="parking"
      prefix={
        <img
          alt="car parking"
          src="addparking.svg"
        />
      }
      route="/rental-profile/parking"
    />
    <Capsule
      buttonLabel="Add Storage"
      expansionPanel={false}
      label="I'll need storage"
      limitReached={false}
      name="storage"
      prefix={
        <img
          alt="storage"
          src="addstorage.svg"
        />
      }
      route="/rental-profile/storage"
    />
  </div>
  <ActionButton
    color="primary"
    disabled={false}
    marginBottom={27}
    marginTop={60}
    onClick={[Function]}
    successGreen={false}
    variant="contained"
  >
    Continue
  </ActionButton>
  <BackLink
    to="/"
  />
</React.Fragment>
`;

exports[`renders ExistingItemsExpansionPanel for parking when there are selected_rental_options with quantity greater than 0 2`] = `
<ExistingParkingOrStorage
  quantity={2}
  rentalOption={
    Object {
      "deposit_amount": null,
      "fee_amount": null,
      "id": 102,
      "included": 1,
      "limit": 6,
      "mandatory": true,
      "monthly_amount": "75.00",
      "name": "Garage (12x9)",
      "payment_time": 10,
      "type": 10,
    }
  }
/>
`;

exports[`renders ExistingItemsExpansionPanel with coapplicants as existing items 1`] = `
<ExistingRoommate
  item={
    Object {
      "first_name": "kreebs",
      "id": 71,
      "is_registered": false,
      "last_milestone": null,
      "last_name": "mcgreebs",
      "phone_number": "(383) 838-4849",
      "status": "Not Started",
    }
  }
  type="co_applicants"
/>
`;
